modules/mcuboot/hw_counters.conf

```diff
@@ -0,0 +1,11 @@
+# This Kconfig fragment is applied to the MCUBoot configuration when
+# we want to enable the hardware security counters for downgrade
+# prevention.
+CONFIG_MCUBOOT_HW_DOWNGRADE_PREVENTION=y
+
+# These NSIB libraries are enabled because they are used by the above
+# option. The above option is defined by MCUBoot's Kconfig sources and
+# it was not clear how to connect the two sets of options aside from
+# enabling them simultaneously as we do here.
```

Comment:
Maybe we could phrase this a little differently.
Now it sounds like it's not possible so we just assign both of them here but I would assume having them depend on one or the other would break other use-cases right?

Reply:
Yeah,

there is no way AFAIK to make the out-of-tree MCUBoot option depend on our options. And it doesn't make sense to make our options depend on the MCUBoot option, because the MCUBoot feature uses our stuff, so the dependency would be in the wrong direction.

And I'm scared to use select because it's not clear if this is safe in this case.

Reply:
Suggestions for how to do or phrase this better are welcome :)

Reply:
Suggestion but maybe it's not that better.
```
# These NSIB libraries are enabled here because they are used by the above
# option. The above option is defined by MCUBoot's Kconfig sources and
# there is currently not a way to sets these options aside from enabling
# them through this KConfig fragment simultaneously as we do here.

Reply:
fixed